components/portal/misc/filters.js

```diff
@@ -85,9 +88,18 @@ define(['angular'], function(angular) {
     })
     .filter('urlToTarget', function() {
       return function(url) {
-        var result = '_self';
+        var result = TARGET_SELF;
         if (url && url.indexOf('//') > -1) {
-          result = '_blank';
+          result = TARGET_BLANK;
+        }
+        return result;
+      };
+    })
+    .filter('targetToRel', function() {
+      return function(target) {
+        var result = '';
+        if (target === TARGET_BLANK) {
```

Comment:
In the interest of having a generically usable `targetToRel` filter, should this be equals ignoring case? Possibly after trimming whitespace from `target`?  Then one could pipe arbitrary `<a` tag `target` attributes in the world through `targetToRel` to get the appropriate `rel` for that anchor?

Reply:
I see trimming being useful, not as much of ignoring case.  But if you're going to do one, it's not hard to do both.

Reply:
we certainly could, but actually we probably shouldn't calculate off of target at all. Since the concern comes from external sites being hijacked, a more foolproof option would be to calculate the rel from the url, like we do in the filter right above this. Or we could straight up just add the rel to every link we create, provided internal pages don't require the referrer

Reply:
Hmm. But it's the `target`, not the URL, that triggers the vulnerability, isn't it? A `target` of `_blank` or `_BLANK` or `_BlAnK` or, browsers being what they are, probably `_blank   ` is what sets up the opportunity for the newly opened tab to mess with the prior no-longer-focus-of-attention tab location, a messing-with that's blocked by the `rel` incantation.

Reply:
ahhh, ok. I have a superficial understanding of the vuln. Yup, I'll clean up the target matching